### PR TITLE
Improve Vtiger-odoo13

### PR DESCRIPTION
**Pull request by dhiren-scs**
Fix the multiple line issue of saleorder,Purchase order and Invoice.

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@hiren-serpentcs pls review

File: vtiger_connector_calendar/models/res_company.py [hunk 1]
```diff
@@ -58,26 +58,13 @@ def sync_vtiger_calendar_event(self):
                     if get_start_date and get_end_date and \
                        get_start_date > get_end_date:
                         calendar_vals.update({'start': str(start_date)})
```
Comment (hiren-serpentcs) [inline vtiger_connector_calendar/models/res_company.py:60]: > str(start_date)

Do we need the typecast here?

File: vtiger_connector_calendar/models/res_company.py [hunk 2]
```diff
@@ -58,26 +58,13 @@ def sync_vtiger_calendar_event(self):
...
-#                         'duration':0.0,
-#                    })
-#                    if get_start_date and get_end_date and\
-#                       get_start_date > get_end_date:
-#                        calendar_vals.update({
-#                             'start': start_date,
-#                             'stop': start_date,
-#                        })
                     else:
                         split_time = str(get_start_time).split(':')
```
Comment (hiren-serpentcs) [inline vtiger_connector_calendar/models/res_company.py:69]: No need to typecast and split, we can directly get hours by get_start_time.hour

File: vtiger_connector_calendar/models/res_company.py (hunk 1 above)
Comment (hiren-serpentcs) [reply vtiger_connector_calendar/models/res_company.py:60]: remove typecasting for date/datetime fields

File: vtiger_connector_invoice/models/res_company.py [hunk 3]
```diff
@@ -52,65 +53,76 @@ def sync_vtiger_invoice(self):
...
             invoice_obj = self.env['account.move']
             partner_obj = self.env['res.partner']
             product_obj = self.env['product.product']
             if result.get('success'):
                 self.delete_existing_invoice(result)
                 for res in result.get('result', []):
+                    io_id = res.get('id')
+                    line_url = company.vtiger_server
+                    line_url += "/restapi/v1/vtiger/default/retrieve?id=" + io_id
+                    line_req = requests.get(line_url, auth=(
```
Comment (hiren-serpentcs) [inline vtiger_connector_invoice/models/res_company.py:66]: Add docstring for why we added the second API call to vtiger

File: vtiger_connector_sales/models/res_company.py [hunk 4]
```diff
@@ -7,6 +7,8 @@
 from datetime import datetime
 from urllib.request import urlopen, Request
 from urllib.parse import urlencode
+import requests
+from requests.auth import HTTPBasicAuth
```
Comment (hiren-serpentcs) [inline vtiger_connector_sales/models/res_company.py:11]: are we using HTTPBasicAuth anywhere on the code?

**Review by hiren-serpentcs — CHANGES_REQUESTED** [review]

File: vtiger_connector_sales/models/res_company.py [hunk 5]
```diff
@@ -150,32 +164,40 @@ def sync_vtiger_sale_Quotes(self):
             if result.get('success'):
                 self.update_existing_sale_Quotes(result)
```
Comment (hiren-serpentcs) [inline vtiger_connector_sales/models/res_company.py:165]: Check update_existing_sale_Quotes and update_existing_sale_order methods, both are having the exact same code

**Review by hiren-serpentcs — CHANGES_REQUESTED** [review]

**Comment by JayVora-SerpentCS** [comment]
use labels wisely @hiren-serpentcs

File: vtiger_connector_partner/models/res_company.py [hunk 6]
```diff
@@ -95,7 +94,7 @@ def sync_vtiger_partner_vendor(self):
                         'name': res.get('vendorname'),
                         'email': res.get('email'),
                         'website': res.get('website'),
```
Comment (hiren-serpentcs) [inline vtiger_connector_partner/models/res_company.py:96]: We can create a new method and that will work on the query data to avoid the code duplication

**Review by hiren-serpentcs — CHANGES_REQUESTED** [review]

**Comment by JayVora-SerpentCS** [comment]
How far we have reached ?

**Comment by JayVora-SerpentCS** [comment]
@chandni-serpentcs pls see

**Pull request closed without merging**